Added a section about not using enums (#2995)
statelyai/xstate: docs/guides/typescript.md

If you want to remove the generated file, just remove the `tsTypes` attribute from your machine and it'll stop being generated.

#### Don't use enums

> Enums are a common pattern in the XState Typescript commmunity. They're often used to declare state names. like this:

Reviewer: This kinda reads like they are common and still useful - maybe using past tense here would be more fitting here?

Author: Done

> Enums are a common pattern in the XState Typescript commmunity. They're often used to declare state names. like this:

Reviewer 2: ```suggestion
Enums were a common pattern used with XState TypeScript. They were often used to declare state names. like this:
```
